src/hooks/useLocalNotifications.tsx

```diff
@@ -57,11 +57,18 @@ const useLocalNotificationsContext = () => {
   const setTransferNotifications = useCallback(
     (notifications: TransferNotifcation[]) => {
       if (!dydxAddress) return;
-      const updatedNotifications = { ...allTransferNotifications };
-      updatedNotifications[dydxAddress] = notifications;
-      setAllTransferNotifications(updatedNotifications);
+      setAllTransferNotifications((currentAllNotifications) => {
```

Comment:
This is a fix for a race condition bug, where `allTransferNotifications` was out dated during status update and can some times override new statuses. Leading to transfer statuses sometime not being saved